src/plugins/sailfishoswizards/sailfishoswizardsplugin.cpp

```diff
@@ -30,6 +31,9 @@ bool SailfishOSWizardsPlugin::initialize(const QStringList &arguments, QString *
 {
     Q_UNUSED(arguments)
     Q_UNUSED(errorString)
+    Utils::globalMacroExpander()->registerVariable("SailfishOSFileWizardCategoryName",
+                                                   tr("Category of a wizard for Sailfish OS files"),
+                                                   []() { return Sfdk::Sdk::osVariant(); });
```

Comment:
Yes, something like that, but
1. done from the SailfishOS plugin
2. only exporting the xxxVariant string
3. using the identifiers as here https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/tools/sfdk/dispatch.cpp#L66 but prefixed with "Sailfish" (e.g. "SailfishOsVariant") and then
4. having something like "Category of a wizard for %{SailfishOsVariant} files" in the json.

Reply:
And 5. changing upstream code with separate commit to be sent upstream :)

Reply:
1 — So, where should we put the variable registration instead of the `sailfishoswizards` plugin?

3 — Do you mean to use `tr("Category of a wizard for %{SailfishOsVariant} files")` instead of `tr("Category of a wizard for Sailfish OS files")`?

5 — Do you mean a separate pull request? Because commit is already separate.

Reply:
> 1 — So, where should we put the variable registration instead of the `sailfishoswizards` plugin?

Done from `MerPlugin::initialize()` should be OK

> 4 — Do you mean to use `tr("Category of a wizard for %{SailfishOsVariant} files")` instead of `tr("Category of a wizard for Sailfish OS files")`?

Sorry, my example was a bit confusing because I confused that message for the macro value, while it is actually the description of the macro. So the JSON side is OK (except the macro name), and the description argument to the `registerVariable` call should be change to as `tr("Category of a wizard for %1 files").arg(Sdk::osVariant())`.

> 5 — Do you mean a separate pull request? Because commit is already separate. Also, do you mean that it should be [only one change inside the `JsonWizardFactory` class](https://github.com/sailfishos/sailfish-qtcreator/pull/260/commits/b483f21d10cb3d223a18050cdb8c1af7809e5fdc#diff-dbc3af13b462fd690c75ef256107c694L614)? Or something else?

I mean separate commit with changes to upstream code, so that you can cherry-pick the one commit and submit it for upstream review. Also the registration of the macros should be done with a separate commit.

Reply:
We added PR about setting a wizard info with variables: https://github.com/sailfishos/sailfish-qtcreator/pull/264